packages/chargebee/src/types.ts

```diff
@@ -0,0 +1,143 @@
+export type VoidFunction = () => void;
+
+type HostedPageId = Nominal<string, 'HostedPageId'>;
+
+type HostedPageType =
+  | 'checkout_new'
+  | 'checkout_existing'
+  | 'update_payment_method'
+  | 'manage_payment_sources'
+  | 'collect_now'
+  | 'extend_subscription'
+  | 'checkout_one_time'
+  | 'pre_cancel'
+  | 'checkout_gift'
+  | 'claim_gift'
+  | 'agreement_pfd';
+
+type HostedPageState =
+  | 'created'
+  | 'requested'
+  | 'succeeded'
+  | 'cancelled'
+  | 'acknowledged';
+
+export type HostedPage = {
+  id: HostedPageId;
+  embed: boolean;
+  created_at: number;
+  expires_at: number;
+  object: 'hosted_page';
+  resource_version: number;
+  state: HostedPageState;
+  type: HostedPageType;
+  updated_at: number;
+  url: string;
+};
+
+type OpenCheckoutLayout =
+  | 'in_app'
+  | 'full_page';
+
+export type OpenCheckoutOptions = {
+  hostedPage: () => HostedPage;
+  close?: VoidFunction;
+  success: (hostedPageId: HostedPageId) => void;
+  error?: VoidFunction;
+  loaded?: VoidFunction;
+  layout?: OpenCheckoutLayout;
+  step?: (currentStep: string) => void;
+};
+
+type ChargebeePortalOptions = {
+  close?: VoidFunction;
+  paymentSourceAdd?: VoidFunction;
+  paymentSourceUpdate?: VoidFunction;
+  paymentSourceRemove?: VoidFunction;
+  loaded?: VoidFunction;
+  visit?: (sectionType: string) => void;
```

Comment:
pretty sure that this section type matches the ones listed here

https://www.chargebee.com/checkout-portal-docs/api.html#getportalsections